tests/trainer/test_trainer.py

```diff
@@ -662,6 +662,11 @@ def check_best_model_has_been_loaded(
         metrics = trainer.evaluate()
         self.assertEqual(metrics[metric], best_value)
 
+    def remove_nan_logs(self, log):
+        for key in list(log.keys()):
+            if log[key] != log[key]:  # Check if the value is NaN
+                del log[key]
```

Comment:
maybe left a comment (on review page) regarding the reason? is this about hpu (as the PR title says HPU failures) or it's something in general (as you mentioned to me internally that `many fsdp failures`